### ISSUE TITLE
CA1024: False positive for extern methods

### ISSUE THREAD

**Issue by skrysmanski** [dotnet/roslyn-analyzers#6031]
### Analyzer

**Diagnostic ID**: [CA1024](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1024): `Use properties where appropriate`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 6.0.0 (Latest)

### Describe the bug

`extern` methods (P/Invoke) should not trigger this rule.

One could argue that P/Invoke methods meet the following exception (from the documentation):

> The method performs a time-consuming operation. The method is perceivably slower than the time that is required to set or get the value of a field.

### Steps To Reproduce

Have a class like this:

```c#
class MyClass
{
        [DllImport("kernel32.dll")]
        public static extern IntPtr GetConsoleWindow();
}
```

### Expected behavior

There should be no CA1024 for the method `GetConsoleWindow()`.

### Actual behavior

CA1024 is reported for the method `GetConsoleWindow()`.

**Comment by Evangelink** [comment]
@buyaa-n I can reproduce the issue. Adding an exception for `DllImport` looks easy but I am wondering if we want to be a bit more global and exclude more cases (e.g. all properties marked with some attribute).

**Comment by skrysmanski** [comment]
Not sure whether you can use `extern` without `[DllImport]` (or the other way around), but I doubt it. 

Under this assumption and if both solution are equally easy to implement, I personally would check for `extern` rather than `[DllImport]`.

**Comment by Evangelink** [comment]
Yep. But attributes are always tricky so maybe we would prefer to have higher False Negatives than False Positives. This being said, my PR with the extern fix is ready.

**Comment by buyaa-n** [comment]
Yes, excluding `extern` methods sounds good to me
